### PR TITLE
format magic help message

### PR DESCRIPTION
**Pull request by brandondutra**
* "%%ml --help" now prints a short description of each command and not the full usage example message.The usage example is moved to argparse's epilog parameter. This also means when printing the help message in DL's CommandParser, we have to make sure the epilogue is printed last.

*  added more spacing for nested yaml parameters

* Show examples of putting line args in the cell in the help message

* help messages are not 100% nice, but getting there.

to do in future PR:

* mltoolbox -> mlworkbench
* wrap help to 80 cols, or at least wrap on word boundaries.

### CODE REVIEW

**Comment by yebrahim** [comment]
It'd be helpful if you can paste an example --help output with and without the change in your PR description.

File: google/datalab/contrib/mltoolbox/commands/_ml.py [hunk 1]
```diff
@@ -59,109 +60,127 @@ def ml(line, cell=None):
...
   transform_parser.add_argument('--analysis', required=True,
                                 help='path of analysis output directory.')
   transform_parser.add_argument('--output', required=True,
                                 help='path of output directory.')
-  transform_parser.add_argument('--prefix', required=True, metavar='NAME',
-                                help='The prefix of the output file name. The output files will ' +
-                                     'be like NAME_00000_of_00005.tar.gz')
+  transform_parser.add_argument(
+      '--prefix', required=True, metavar='NAME',
+      help='The prefix of the output file name. The output files will be like '
```
Comment (yebrahim) [inline google/datalab/contrib/mltoolbox/commands/_ml.py]: missing `+` at the end of this line?

File: google/datalab/utils/commands/_commands.py [hunk 2]
```diff
@@ -56,15 +56,20 @@ def exit(self, status=0, message=None):
...
     """Override help doc to add cell args. """
 
-    orig_help = super(CommandParser, self).format_help()
     if not self._cell_args:
-      return orig_help
+      return super(CommandParser, self).format_help()
+    else:
+      # Print the standard argparse info, the cell arg block, and then the epilog
+      epilog = self.epilog
+      self.epilog = None
```
Comment (yebrahim) [inline google/datalab/utils/commands/_commands.py]: What does this achieve?

**Comment by brandondutra** [comment]
example of %%ml --help

now: https://screenshot.googleplex.com/x7mRyibBRYQ.png
before: https://screenshot.googleplex.com/dB6ejWGYAKm.png

File: google/datalab/utils/commands/_commands.py (hunk 2 above)
Comment (brandondutra) [inline google/datalab/utils/commands/_commands.py]: The epilog should be printed list. But the cell help block was being printed last. Updated comment.

File: google/datalab/contrib/mltoolbox/commands/_ml.py (hunk 1 above)
Comment (brandondutra) [inline google/datalab/contrib/mltoolbox/commands/_ml.py]: Let me remove all the '+' as they are not needed.

**Pull request merged**